src/java.base/windows/classes/sun/nio/ch/WEPoll.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2020, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.nio.ch;
+
+import java.io.IOException;
+import jdk.internal.misc.Unsafe;
+
+/**
+ * Provides access to wepoll.
+ */
+class WEPoll {
+    private static final Unsafe UNSAFE = Unsafe.getUnsafe();
+
+    private WEPoll() { }
+
+    /**
+     * typedef union epoll_data {
+     *     void *ptr;
+     *     int fd;
+     *     uint32_t u32;
+     *     uint64_t u64;
+     *     SOCKET sock;   // Windows specific
+     *     HANDLE hnd;    // Windows specific
+     *  } epoll_data_t;
+     *
+     * struct epoll_event {
+     *     uint32_t events;
+     *     epoll_data_t data;
+     * }
+     */
+    private static final int SIZEOF_EPOLLEVENT   = eventSize();
+    private static final int OFFSETOF_EVENTS     = eventsOffset();
+    private static final int OFFSETOF_FD         = dataOffset();
+
```

Comment:
The naming here seems a bit inconsistent. The field in the union is called 'sock' but the constant value here refers to OFFSETOF_FD. Would OFFSETOF_SOCK be better? Also, is the getDescriptor() method needed as it doesn't seem to be called? What drew my attention to it is that 'fd' is an int size, but socket handles are long sized (which is how they are accessed through `getSocket`

Reply:
It's a union and they have the same offset. But okay, there may not be a need to provide access to data.fd, in which case getDescriptor could be removed and the private field renamed.